packages/dashboard-core-plugins/src/panels/NotebookPanel.tsx

```diff
@@ -343,24 +348,21 @@ class NotebookPanel extends Component<NotebookPanelProps, NotebookPanelState> {
     this.initTabClasses(tab);
   }
 
-  // override glContainer.close() with a custom closure that checks if needs saving
   initTabCloseOverride() {
     const { glContainer } = this.props;
-    const close = glContainer.close.bind(glContainer);
-    glContainer.close = (isOverwrite = false) => {
-      if (isOverwrite) {
-        close();
+    glContainer.beforeClose((options?: CloseOptions) => {
+      if (options !== undefined && options.force === true) {
```

Comment:
You can use optional chaining to shorten this, e.g.:
```
if (options?.force) {
  return true;
}
```

Reply:
Getting this ESlint error: Unexpected any value in conditional. An explicit comparison or type cast is required. Is this okay?
```
if (options?.force === true) {
  return true;
}
```

Reply:
Yup that's good